File: src/389_find_the_difference/FindTheDifference.java
public class FindTheDifference {
    public char findTheDifference(String s, String t) {
        int sSum = 0, tSum = 0;
        for (int i = 0; i < t.length(); i++) {
            if (i < t.length() - 1) {
                sSum += s.charAt(i);
            }
            tSum += t.charAt(i);
        }
        return (char) (tSum - sSum);
    }

    public static void main(String[] args) {
        FindTheDifference sol = new FindTheDifference();
        System.out.println(sol.findTheDifference("abcd", "abcde") + " | e");
        System.out.println(sol.findTheDifference("", "y") + " | y");
        System.out.println(sol.findTheDifference("a", "aa") + " | a");
        System.out.println(sol.findTheDifference("ae", "aea") + " | a");
        System.out.println(sol.findTheDifference("xyz", "xzyq") + " | q");
    }
}

File: src/389_find_the_difference/find_the_difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        sSum = 0
        tSum = 0

        lenT = len(t)

        for i in range(lenT):
            if i < lenT - 1:
                sSum += ord(s[i])

            tSum += ord(t[i])

        return chr(tSum - sSum)


def test():
    sol = Solution()
    print(sol.findTheDifference("abcd", "abcde"), "| e")
    print(sol.findTheDifference("", "y"), "| y")
    print(sol.findTheDifference("a", "aa"), "| a")
    print(sol.findTheDifference("ae", "aea"), "| a")
    print(sol.findTheDifference("xyz", "xzyq"), "| q")


if __name__ == "__main__":
    test()
